dialer.go

```diff
@@ -86,6 +95,12 @@ type Dialer struct {
 	// dialFunc is the function used to connect to the address on the named
 	// network. By default it is golang.org/x/net/proxy#Dial.
 	dialFunc func(cxt context.Context, network, addr string) (net.Conn, error)
+
+	useIAMAuthN bool
+	tokenSource oauth2.TokenSource
```

Comment:
should this be `iamTokenSource` for consistency with cloud sql?: https://github.com/GoogleCloudPlatform/cloud-sql-go-connector/blob/main/dialer.go#L97